### PR TITLE
Jormun: Update status

### PR DESCRIPTION
**Pull request by AurelienLP**

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/__init__.py [hunk 1]
```diff
@@ -49,8 +49,18 @@ def add_common_status(response, instance):
         response['status']['realtime_proxies'].append(realtime_proxy.status())
 
     response['status']['street_networks'] = []
-    for sn in instance.get_all_street_networks():
-        response['status']['street_networks'].append(sn.status())
+    street_networks = instance.get_all_street_networks()
+    # street_networks is a list if it comes from the Json
+    if type(street_networks) == list:
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/__init__.py]: `isinstance` is better than `type`

**Review by xlqian — APPROVED** [review]

**Comment by AurelienLP** [comment]
Jenkins, test this please.

File: source/jormungandr/jormungandr/interfaces/v1/__init__.py (hunk 1 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/interfaces/v1/__init__.py]: it's best to check that it's some abstract type with https://docs.python.org/2.7/library/collections.html#collections-abstract-base-classes

File: source/jormungandr/jormungandr/interfaces/v1/serializer/status.py [hunk 2]
```diff
@@ -119,6 +119,11 @@ class StreetNetworkSerializer(OutsideServiceCommon):
     def get_timeout(self, obj):
         return obj.get('timeout', None)
 
+    backend_class = MethodField(schema_type=str, display_none=False)
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/status.py]: You should put it above `get_timenout()` to improve readability

**Review by kinnou02 — APPROVED** [review]

File: source/jormungandr/jormungandr/interfaces/v1/__init__.py (hunk 1 above)
Comment (AurelienLP) [reply source/jormungandr/jormungandr/interfaces/v1/__init__.py]: Here, I know for certain that this is a list. Should I still check if it is an abstract class?

File: source/jormungandr/jormungandr/interfaces/v1/serializer/status.py (hunk 2 above)
Comment (AurelienLP) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/status.py]: `get_backend_class()` to or just the field?

File: source/jormungandr/jormungandr/interfaces/v1/__init__.py (hunk 1 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/interfaces/v1/__init__.py]: it'r better to check for an abstract class as if you change implementation of `get_all_street_networks()` to return another kind of container, like a set or some kind of readonly list by example your code will still works.

File: source/jormungandr/jormungandr/interfaces/v1/serializer/status.py (hunk 2 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/status.py]: just the field, to group them

File: source/jormungandr/jormungandr/interfaces/v1/__init__.py [hunk 3]
```diff
@@ -49,8 +51,17 @@ def add_common_status(response, instance):
         response['status']['realtime_proxies'].append(realtime_proxy.status())
 
     response['status']['street_networks'] = []
-    for sn in instance.get_all_street_networks():
-        response['status']['street_networks'].append(sn.status())
+    street_networks = instance.get_all_street_networks()
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/interfaces/v1/__init__.py]: The output of a method should always be the same type, whatever the configuration is

File: source/jormungandr/jormungandr/street_network/tests/streetnetwork_backend_manager_test.py [hunk 4]
```diff
@@ -261,3 +262,26 @@ def wrong_streetnetwork_backend_test():
     manager._update_config("instance")
     assert not manager._streetnetwork_backends_by_instance_legacy
     assert not manager._streetnetwork_backends
+
+
+class FakeInstance:
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/street_network/tests/streetnetwork_backend_manager_test.py]: This should inherit from Instance

**Review by kinnou02 — CHANGES_REQUESTED** [review]

**Review by kinnou02 — APPROVED** [review]

**Review by xlqian — APPROVED** [review]

**Pull request merged**